### DISCUSSION TITLE
Bottom Sheet

### DISCUSSION THREAD

**Discussion by codeanees** [CommunityToolkit/Maui#143] [New Feature Discussions]
Recently I came across this blog and I love to port or include this feature in Maui toolkit
https://xamarinhowto.com/xamarin-forms-lightweight-bottom-sheet-modal/

**Comment by TheCodeTraveler** [comment]
Neat! We [recently added `Snackbar`](https://github.com/CommunityToolkit/Maui/pull/5) and are working on [Popup](https://github.com/CommunityToolkit/Maui/issues/114). 

I could see `BottomModalSheet` being added to the `CommunityToolkit.Maui.Alerts` namespace.

Let's put this on hold until Popup has been merged, and then see how we can leverage Popup's implementation for `BottomModalSheet`. 

After we merge Popup, @codeanees, I'd love for you to provide an API design!

**Reply by TheCodeTraveler** [reply]
👆 @pictos Just FYI since you're working on Popup

**Comment by TheCodeTraveler** [comment]
Notes [discussed in January's Standup](https://github.com/CommunityToolkit/Maui/discussions/143):
- Can we add functionality so that we can "fly in" the sheet from the sides and top as well?
- Can we leverage [Xamarin Community Toolkit's SideMenuView](https://github.com/xamarin/XamarinCommunityToolkit/tree/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView)?
- Can we leverage existing code from Popup, Toast and Snackbar?

**Comment by eramrit78** [comment]
is bottomsheet added in maui alerts?. nothing any thing found in docs. bottom sheet is a most demanding feature in today apps.

**Reply by TheCodeTraveler** [reply]
It is not. Would like to propose an API design for BottomSheet?

**Reply by VladislavAntonyuk** [reply]
Possible api: https://github.com/CommunityToolkit/Maui/discussions/454

**Reply by jfversluis** [reply]
Maybe we should contact @naweed and see if we can leverage some of his code: https://blogs.xgenoapps.com/post/2022/07/23/maui-bottom-sheet and https://github.com/naweed/Maui.Controls.BottomSheet

**Reply by VladislavAntonyuk** [reply]
ideally, if we can reuse the popup, but it needs to support dynamic height.

**Reply by IeuanWalker** [reply]
@jfversluis This can be achieved using Rg.Popups maui replacement. But can be completely customised by the user. I only have one issue with it atm - https://github.com/LuckyDucko/Mopups/issues/12

But you don't need to use a PanGestureRecognizer, it all works fine as a bottom sheet.

Another example, i used it to create my own picker - https://github.com/LuckyDucko/Mopups/discussions/8#discussioncomment-3136608

**Comment by VladislavAntonyuk** [comment]
I propose to implement it similar to SnackBar and Toast using native UISheetPopupController and BottomSheetDialog
https://vladislavantonyuk.azurewebsites.net/articles/Creating-a-bottom-sheet-using-.NET-MAUI

**Comment by durandt** [comment]
A bottom sheet would be very nice to have, it's part of many mobile applications design. 

The @naweed implementation is neat but I didn't find an easy way to make the bottom sheet hide the bottom tabs when using it within a Shell with tabs.

Trouble with the UISheetPresentationController-approach is that it is only available from SDK 15 and customization from SDK 16 (and about 5% of my users are stuck on iOS 14 for whatever reason beyond my understanding since no devices were dropped by iOS 15).

I will take a bet and go forward working from the Popup implementation and try to make a Bottom Sheet implementation for our project (Android & iOS, Bottom Sheet only (no top/left/right), Three pre-defined sizes (less dynamic content stuff)) and will see if we can make something that could contribute back to the Community Toolkit.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889

**Reply by Cybrosys** [reply]
Could you clarify? The timestamp does not mention Bottom Sheet but instead mentions that a discussion needs 7 or more votes not to be closed, which this one has, but it was closed.

**Reply by durandt** [reply]
I was also a bit disappointed & thought it was a bit unclear but in the end I understand they need ways to triage discussions.
My understanding was that the 7-threshold was only a suggestion to change the suggested policy of 10 votes, and that the policy that was clubbed was 10votes.
